### PR TITLE
James 1693

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJamesServerTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                        new TestJMAPServerModule(GetMessageListMethod.DEFAULT_MAXIMUM_LIMIT),
+                        new AbstractModule() {
+                    
+                    @Override
+                    protected void configure() {
+                    }
+                    
+                    @Provides
+                    @Singleton
+                    com.datastax.driver.core.Session provideSession(CassandraModule cassandraModule) {
```
Comment (aduprat) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJamesServerTest.java]: You may not need the FQN

File: server/container/guice/memory-guice/README.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * james-server-memory-guice-${version}.lib
+
+== Running
+
+=== James Launch
+
+To run james, you have to create a directory containing required configuration files.
+
+A sample directory (appropriately named sample-directory) is provided with some
+default value you may need to replace.
```
Comment (aduprat) [inline server/container/guice/memory-guice/README.adoc]: s/value/values

File: server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/servers/CassandraJmapServerModule.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    protected void configure() {
+        install(new TestElasticSearchModule(embeddedElasticSearch));
+        install(new TestFilesystemModule(temporaryFolder));
+        install(new TestJMAPServerModule(LIMIT_TO_3_MESSAGES));
+        bind(EmbeddedCassandra.class).toInstance(cassandra);
+    }
+    
+    @Provides
+    @Singleton
+    com.datastax.driver.core.Session provideSession(CassandraCluster initializedCassandra) {
```
Comment (aduprat) [inline server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/servers/CassandraJmapServerModule.java]: FQN

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/MemoryGetMessageListMethodTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    protected GuiceJamesServer<?> createJmapServer() {
+        return new GuiceJamesServer<>(new TypeLiteral<InMemoryId>(){})
+                    .combineWith(MemoryJamesServerMain.inMemoryServerModule)
+                    .overrideWith(new MemoryJmapServerModule(temporaryFolder));
+    }
+    
+    @Override
+    protected void await() {
```
Comment (aduprat) [inline server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/MemoryGetMessageListMethodTest.java]: Love this !

File: server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    public static class MemoryMailQueue implements MailQueue {
+
+        private final LinkedBlockingDeque<MemoryMailQueueItem> mailItems;
+        private final MailQueueItemDecoratorFactory mailQueueItemDecoratorFactory;
+        private final String name;
+
+        public MemoryMailQueue(String name,MailQueueItemDecoratorFactory mailQueueItemDecoratorFactory) {
+            mailItems = new LinkedBlockingDeque<>();
```
Comment (aduprat) [inline server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java]: You may add `this.` in order to keep the pattern of next lines.

File: server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueModule.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    public MemoryMailQueueModule(TypeLiteral<Id> type) {
+        guiceGenericType = new GuiceGenericType<>(type);
+    }
+    
+    @Override
+    protected void configure() {
+        bind(MailQueueFactory.class).to(MemoryMailQueueFactory.class);
+        bind(MailQueueItemDecoratorFactory.class)
+        .to(guiceGenericType.newGenericType(PostDequeueDecoratorFactory.class))
```
Comment (aduprat) [inline server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueModule.java]: indent

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJamesServerTest.java (hunk 1 above)
Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJamesServerTest.java]: ok

File: server/container/guice/memory-guice/README.adoc (hunk 2 above)
Comment (mbaechler) [inline server/container/guice/memory-guice/README.adoc]: ok

File: server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/servers/CassandraJmapServerModule.java (hunk 3 above)
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/servers/CassandraJmapServerModule.java]: ok

File: server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java (hunk 5 above)
Comment (mbaechler) [inline server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java]: ok

File: server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueModule.java (hunk 6 above)
Comment (mbaechler) [inline server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueModule.java]: ok

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java [hunk 7]
```diff
@@ -19,63 +19,39 @@
...
+    public static final TypeLiteral<CassandraId> cassandraId = new TypeLiteral<CassandraId>(){};
+    public static final Module cassandraServerModule = Modules.combine(
+        new CassandraUsersRepositoryModule(),
+        new CassandraDomainListModule(),
+        new CassandraRecipientRewriteTableModule(),
+        new CassandraMailboxModule(),
+        new CassandraSessionModule(),
+        new ElasticSearchMailboxModule(),
+        new QuotaModule(),
+        new ActiveMQQueueModule<>(cassandraId));
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java]: Nice !

**Comment by chibenwa** [comment]
As these file were copied I cannot comment anywhere but here.

IMO, jcr-repository.xml can be removed from sample configuration
fakemailrepositorystore.xml can be removed from JMAP integration resources
I think it is even the same for RRT config file in sample configuration (at least class attribute can be removed). Idem for domain list and users config files every where...

Comments applies on https://github.com/linagora/james-project/pull/191/commits/22795f447b310d4d20a028481c1a2f8f26fbbed3#diff-c592cb44ea3113b7dc66f704720c0368R48

File: server/container/guice/memory-guice/src/test/resources/mailetcontainer.xml [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+        <threads>20</threads>
+    </spooler>
+
+    <processors>
+        <processor state="root" enableJmx="false">
+            <mailet match="All" class="PostmasterAlias"/>
+            <mailet match="RelayLimit=30" class="Null"/>
+            <!-- Hook on sievemanager@james.linagora.com
+                 Mail send to this address will get interpreted with SIEVE Manage -->
+            <mailet match="RecipientIs=sievemanager@james.linagora.com" class="ToProcessor">
```
Comment (chibenwa) [inline server/container/guice/memory-guice/src/test/resources/mailetcontainer.xml]: Maybe we can simplify this configuration file to remove all non needed mailets (saves time on Camel startup)

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/fakemailrepositorystore.xml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/fakemailrepositorystore.xml]: File not needed ?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/mailetcontainer.xml [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+    <spooler>
+        <threads>20</threads>
+    </spooler>
+
+    <processors>
+        <processor state="root" enableJmx="false">
+            <mailet match="All" class="PostmasterAlias"/>
+            <mailet match="RelayLimit=30" class="Null"/>
+            <!-- Hook on sievemanager@james.linagora.com
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/mailetcontainer.xml]: This commentseems unrelevent as the concerned manageSieve hook was removed...

**Comment by chibenwa** [comment]
test.properties + test.xml are not needed as resources of jmap-integration-testing-common

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/resources/fakemailrepositorystore.xml [hunk 11]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/resources/fakemailrepositorystore.xml]: I think this file is not needed

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/resources/mailetcontainer.xml [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+    <spooler>
+        <threads>20</threads>
+    </spooler>
+
+    <processors>
+        <processor state="root" enableJmx="false">
+            <mailet match="All" class="PostmasterAlias"/>
+            <mailet match="RelayLimit=30" class="Null"/>
+            <!-- Hook on sievemanager@james.linagora.com
+                 Mail send to this address will get interpreted with SIEVE Manage -->
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/resources/mailetcontainer.xml]: Comment irrelevant ?

**Comment by chibenwa** [comment]
Appart uneeded files in configuration (test resources) and unneeded classes for server-data configuration, this is a :+1: for me.

Having only the needed configuraion files seems important to me, as it makes it more clear on which information James relies really, without to look at the code.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/mailetcontainer.xml (hunk 10 above)
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/mailetcontainer.xml]: not related, we should make a global review of config file at another time

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/resources/mailetcontainer.xml (hunk 12 above)
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/resources/mailetcontainer.xml]: not related, we should make a global review of config file at another time

**Comment by mbaechler** [comment]
I removed unneeded files but won't change files content right now.
We should do a careful review of config file as a task of its own

**Comment by aduprat** [comment]
:+1:

**Comment by chibenwa** [comment]
:+1:

**Pull request merged**